### ISSUE TITLE
Symbol’s function definition is void: org-clock-convenience-org-hide-drawer-toggle 

### ISSUE THREAD

**Issue by mbarton98** [dfeich/org-clock-convenience#14]
I got the updated org-clock-convenience package from melpa.org today and when I try to use the functions in Org Agenda to fix a clocked item, I get:
`org-clock-convenience-open-if-in-drawer: Symbol’s function definition is void: org-clock-convenience-org-hide-drawer-toggle`

(Edit - Just to be clear, it has been working fine and I use it almost daily.)

I see the code that is intended to run, but I'm not sure why it does not in my case.

Here is how I load the package:
```
(use-package org-clock-convenience
  :ensure t
  :after (org)
  :bind (:map org-agenda-mode-map
	      ("<S-up>" . org-clock-convenience-timestamp-up)
	      ("<S-down>" . org-clock-convenience-timestamp-down)
	      ("C-M-S-s-g" . org-clock-convenience-fill-gap)
	      ("é" . org-clock-convenience-fill-gap-both)
	      ))
```

I believe that my system is using Org from the Emacs master tree as shown in the version strings below:
Org mode version 9.5.3 (release_9.5.3-3-gd54104 @ /Applications/Emacs.app/Contents/Resources/lisp/org/)

GNU Emacs 29.0.50 (build 1, x86_64-apple-darwin21.4.0, NS appkit-2113.40 Version 12.3.1 (Build 21E258)) of 2022-05-01
compiled from master be3267eb346745d73cfb627c6e962e261a51d6d2

Here is where you check for Org version older than 9.4. When I evaluate that test I get nil, since 9.5.3 is newer, as expected.
```
(eval-when-compile
  (if (< emacs-major-version 25)
      (defmacro org-clock-convenience-save-mark-and-excursion (&rest body)
        `(save-excursion ,@body))
    (defalias 'org-clock-convenience-save-mark-and-excursion
      'save-mark-and-excursion))
  (if (version< org-version "9.4")
      (defmacro org-clock-convenience-org-hide-drawer-toggle (&optional
                                                              force no-error
                                                              element)
        `(org-flag-drawer nil))
    (defalias 'org-clock-convenience-org-hide-drawer-toggle
      'org-hide-drawer-toggle)))
```
If I evaluate the else part of the if, then this defines the symbol and the issue is resolved for that session.
```
(defalias 'org-clock-convenience-org-hide-drawer-toggle
      'org-hide-drawer-toggle)
```

If I look at help for one of the functions, I see that I have it native comp:
```
org-clock-convenience-fill-gap is an interactive native compiled Lisp
function in ‘org-clock-convenience.el’.
```

**Comment by mbarton98** [comment]
This might be related to the issue I had a while ago with byte compiled code. I was able to resolve this by deleting the elc file in the elpa package folder and also deleted the corresponding file from the eln cache folder. I have restarted Emacs twice to be sure and it works fine now.

So the help shows now that it is not compiled.
```
org-clock-convenience-fill-gap is an autoloaded interactive Lisp
function in ‘org-clock-convenience.el’.
```

Feel free to close this issue.

**Comment by dfeich** [comment]
Hi. Thanks again for an excellent bug-report with all the information for immediately identifying the problem. Prodded by the output of linter runs I had decided to define the compatibility functions with `defalias`. But since this was only executed during compilation (I used it inside an `eval-when-compile`), the alias was not defined when the resulting compiled elc file was loaded.

So, when you deleted the elc file and directly loaded the basic *.el file, everything was fine.

I replaced the defalias invocations by defmacros so that the correct code will end up in the elc file.
Also, your report raised my attention to not just test  the *.el file, but the compiled file.

Best regards
